sunpy/coordinates/frames.py

```diff
@@ -847,3 +856,99 @@ class GeocentricEarthEquatorial(SunPyBaseCoordinateFrame):
     Aberration due to Earth motion is not included.
     """
     equinox = TimeFrameAttributeSunPy(default=_J2000)
+
+
+class BaseMagnetic(SunPyBaseCoordinateFrame):
+    """
+    Base class for frames that rely on the Earth's magnetic model (MAG, SM, and GSM).
+
+    This class is not intended to be used directly and has no transformations defined.
+    """
+    magnetic_model = Attribute(default='igrf13')
+
+    @property
+    def _lowest_igrf_coeffs(self):
+        if not self.magnetic_model.startswith("igrf"):
+            raise ValueError
+
+        igrf_file = download_file("https://www.ngdc.noaa.gov/IAGA/vmod/coeffs/"
+                                  f"{self.magnetic_model}coeffs.txt", cache=True)
```

Comment:
Can we use the data manager here instead?

Reply:
Er, no, because this doesn't follow the model of our data manager:

- The required file is one of several possibilities, but which one is actually needed is resolved only at runtime.  I wouldn't want to blanket require all of the possibilities, so I'd have to jump through some hoops with subfunctions in order to avoid downloading what is not needed.
- I want to be able to immediately support `igrf14coeffs.txt` when it is comes out in 2025, without needing to issue a bugfix or greater release to do so.  I can't provide the data manager with a hash to a file that doesn't exist yet.

Reply:
I think the `sunpy.data.cache` part of "data manager" could be used, not the decorator with the hashes.

Reply:
Using `cache` now